### DISCUSSION TITLE
dap --build-flags via vscode launch.json, how?

### DISCUSSION THREAD

**Discussion by aathan** [go-delve/delve#3461] [Q&A]
I'm having a heck of a time trying to figure out how to pass --build-flags to delve via vscode's launch.json. I'm using vscode on  mac os x connecting to a remote linux environment. I need to pass `-mod=mod` so that the `replace =>` in go.mod of the project is honored (vs build/debug against the vendor'ed version of the replaced modules). If I specify `dlvFlags` in launch.json, I get the `Warning: build flags ignored with dap; specify via launch/attach request instead`

This cryptic message appears to be saying something like "you need to pass these options to dlv via the DAP protocol's launch/attach request instead of specifying them in dlvFlags"... ok.  How?!

EDIT:  Ok, so having read the delve source code it appears that the DAP decoder for the `launch` DAP command can receive an `arguments` json etnry, but now ... how the heck do I get vscode to pass this?!

See #3462

**Comment by firelizzard18** [comment]
Have you tried `"go.buildFlags": ["-mod=mod"]`? Is that not passed to delve?
